stdlib/float.ml

```diff
@@ -271,7 +271,8 @@ module Array = struct
     else if length a2 = 0 then unsafe_sub a1 0 l1
     else append_prim a1 a2
 
-  let fill a ofs len v =
+  (* inlining exposes a float-unboxing opportunity for [v] *)
+  let[@inline] fill a ofs len v =
```

Comment:
I did my tests using a version where `check` is also marked as` [@inline]` (but I did not look at whether it is already inlined by default).

Btw, the implementation of `check` in OCaml seems significantly slower than the built-in array bound  check (for regular get/set), perhaps because the built-in check uses a single unsigned comparison -- `jbe` -- instead of 2 tests.  I wonder if we could somehow expose a %-primitive for such unsigned comparison and benefit from it in this `check` function.

Reply:
I find it easy to justify manually inlining `fill`, because it exposes an unboxing opportunity. On the other hand deciding whether to inline `check` should be the job of the optimizer, I think.

I'm not an expert on unsigned comparison tricks, but here we want to check that `ofs` and `length` are both non-negative and that their sum is in-bound (without overflows), so I think that there is more work to do than just the usual bound-check for one index, and so I'm not sure that it has a more compact formulation.

Reply:
You could save one comparison by replacing `ofs + len < 0 || ofs + len > length a` with `ofs + len >u length a`, but I agree that you still need to check for non-negativity of the two arguments independently so it's not clear that it would be very beneficial.
(I'm interested in exposing unsigned comparison primitives for unrelated reasons though)

Reply:
I don't know if the gain is worth the trouble, but checking if any of two signed integers is negative could be done with a bitwise `or` followed by a single sign check + conditional jmp (rather than 2 comparisons and 2 jumps).